### PR TITLE
Generate Modus Icons based on `@trimble-oss/modus-icons` package

### PR DESCRIPTION
**Pull request by ryan-henness-trimble**
### Overview

This changeset streamlines the integration of Modus Icons into the Stencil project. Previously, including these icons was a tedious process, requiring manual addition of SVG files. This resulted in incomplete icon coverage, and negative impacts on both UX designers and development efficiency.

Instead of the manual inclusion, icons are now automatically generated through a `prebuild` script. This script uses the SVG files provided by the `@trimble-oss/modus-icons` package as the source of truth, ensuring a comprehensive and up-to-date icon set, automatically.

### Key Changes

- **Automated Icon Generation:** The `prebuild` script eliminates the need for manual SVG file additions.
- **Component Renaming:** The `<IconMap />` component is now `<ModusIconMap />`, reflecting its specific functionality and reducing conflicts with the Map icon naming.
- **Naming Consistency:** Icon naming aligns with the Modus Icons package standards. For instance, the "Alarm On" icon, previously referenced as 'alarm-on' in kebab case, will now be identified as `alarm_on` in line with Modus naming conventions.

### Generated Files (that are git-ignored)

[EJS](https://ejs.co/) templates are used to generate the TypeScript files for the following:

- `src/icons/generated-icons/*`: Functional components for all of the Modus Icons
- `src/icons/ModusIconMap.tsx`: A functional component used to map strings to the generated icon components
- `src/icons/ModusIconUtilities.ts`: A file containing utilities that provide all of the Modus Icon names

### Benefits

- **Effortless Icon Updates:** The new process is automatic and efficient, no more manual updates!
- **Ease of Use:** Matching our naming conventions with those of the Modus Icons package simplifies the process for consumers and developers. When integrating icons into web components, the straightforward naming reduces confusion and enhances usability.

### Additional Notes
- The `icons` folder has been moved out of the `components` folder as that folder is reserved for our Stencil web components
- You'll see that there is still a `src/icons/svgs` folder that contains some of the previously manually added SVG icon components. These are leftover to help aid components that were depending on the SVGs directly, instead of using the old `<IconMap />`

### Alternative Approach Considerations
- An alternative approach involving the import of Modus Icon CSS/font via CDN was explored. However, I was not able to resolve the issues presented by sharing imports through the shadow DOM, leading to the the current method.
- I considered making the `prebuild` a `postinstall` instead, but that script runs on the consumer's machine after installing the library so I strayed away from that. If we were to change to `postinstall`, the generation script will need to be included in the output of the package

## Breaking Change
- Strings passed into MWC as Modus Icon names need to match the `@trimble-oss/modus-icons` naming convention

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Documentation update

### CODE REVIEW

File: stencil-workspace/package.json [hunk 1]
```diff
@@ -36,7 +36,8 @@
     "prettier": "prettier --write \"storybook/stories/**/*.*\" \"src/**/*.*\" --config ../.prettierrc.json --ignore-path ../.prettierignore",
     "full": "npm ci && npm run lint && npm run build && npm run test && npm run ci-storybook",
     "ci-storybook": "cd storybook && npm ci && npm run build-storybook",
-    "start-storybook": "npm run build && cd storybook && npm run storybook"
+    "start-storybook": "npm run build && cd storybook && npm run storybook",
+    "prebuild": "cd generate-icons && tsc generate-icons.ts && node generate-icons.js"
```
Comment (ryan-henness-trimble) [inline stencil-workspace/package.json:40]: It seems there is an error in the Netlify PR build for this command. I'm not sure how to resolve this

Comment (ryan-henness-trimble) [reply stencil-workspace/package.json]: Resolved through fixing relative pathing

**Review by cjwinsor — APPROVED** [review]

**Comment by cjwinsor** [comment]
@coliff Looks like you're code owner here, so we'll need your final approval to merge.

**Review by cjwinsor — APPROVED** [review]

**Review by coliff — APPROVED** [review]

**Review by coliff — APPROVED** [review]

**Review by coliff — APPROVED** [review]

**Comment by ryan-henness-trimble** [comment]
@coliff @cjwinsor Thank you guys for all of your hard work reviewing the PRs

**Comment by cjwinsor** [comment]
> @coliff @cjwinsor Thank you guys for all of your hard work reviewing the PRs

@ryan-henness-trimble Thank you for the excellent submission

**Pull request merged**